===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ managed_track_session_inlock:n$3 == { 0 } }) free(*(managed_track_session_inlock:value)) at 5 (line 1605, column 7) }
===== Actual patch in diff format =====
@@ -1603,6 +1603,7 @@ managed_track_session_inlock (p11_dict *sessions,
 	return_val_if_fail (value != NULL, CKR_HOST_MEMORY);
 
 	if (!p11_dict_set (sessions, key, value))
+		free(value);
 		return_val_if_reached (CKR_HOST_MEMORY);
 
 	return CKR_OK;
